lib/patreon/api.rb

```diff
@@ -10,28 +11,33 @@ def initialize(access_token)
       @access_token = access_token
     end
 
-    def fetch_user(includes=nil, fields=nil)
-      get_json(Utils::JSONAPI::URLUtil.build_url('current_user',includes,fields))
+    def fetch_user(opts = {})
+      get_parse_json(Utils::JSONAPI::URLUtil.build_url('current_user', opts[:includes], opts[:fields]))
     end
 
-    def fetch_campaign(includes=nil, fields=nil)
-      get_json(Utils::JSONAPI::URLUtil.build_url('current_user/campaigns',includes,fields))
+    def fetch_campaign(opts = {})
+      get_parse_json(Utils::JSONAPI::URLUtil.build_url('current_user/campaigns', opts[:includes], opts[:fields]))
     end
 
-    def fetch_campaign_and_patrons(includes=nil, fields=nil)
+    def fetch_campaign_and_patrons(opts = {})
       includes ||= Schemas::Campaign.default_relationships + [Schemas::Campaign::Relationships::PLEDGES]
-      fetch_campaign(includes, fields)
+      fetch_campaign(opts)
     end
 
-    def fetch_page_of_pledges(campaign_id, page_size, cursor=nil, includes=nil, fields=nil)
-      url = "campaigns/#{campaign_id}/pledges"
-      url += "?page%5Bcount%5D=#{CGI::escape(page_size.to_s)}"
-      url += "&page%5Bcursor%5D=#{CGI::escape(cursor.to_s)}" if cursor
-      get_json(Patreon::Utils::JSONAPI::URLUtil.build_url(url,includes,fields))
+    def fetch_page_of_pledges(campaign_id, page_size, opts = {})
+      params = { "page[count]" => page_size.to_s }
+      params["page[cursor"] = opts[:cursor] if opts[:cursor]
```

Comment:
Is this quoting and bracketing in the right order?

Reply:
no, no it is not. don't trust any of this until we've got some tests going